checker/bin/wpi.sh

```diff
@@ -291,18 +293,17 @@ elif [ "${has_java16}" = "yes" ]; then
 fi
 configure_and_exec_dljc "$@"
 
-if [ "${has_java11}" = "yes" ] && [ "${WPI_RESULTS_AVAILABLE}" != "yes" ]; then
-    # If running under Java 11 fails, try Java 8.
-    if [ "${has_java8}" = "yes" ]; then
-      export JAVA_HOME="${JAVA8_HOME}"
-      echo "couldn't build using Java 11; trying Java 8"
-      configure_and_exec_dljc "$@"
-    fi
+# If running under Java 8 failed and Java 11 is also available, then try running again under Java 11.
+if [ "${has_java11}" = "yes" ] && [ "${has_java8}" = "yes" ] && [ "${WPI_RESULTS_AVAILABLE}" != "yes" ]; then
+    export JAVA_HOME="${JAVA11_HOME}"
+    echo "couldn't build using Java 8; trying Java 11"
+    configure_and_exec_dljc "$@"
 fi
 
-if [ "${has_java11}" = "yes" ] && [ "${has_java8}" = "yes" ] && [ "${WPI_RESULTS_AVAILABLE}" != "yes" ]; then
-    # If running under Java 11 and Java 8 fails, try Java 16.
-    if [ "${has_java16}" = "yes" ]; then
+# If running under Java 8 and/or 11 failed and Java 16 is available, then try Java 16.
+if [ "${has_java16}" = "yes" ] && [ "${WPI_RESULTS_AVAILABLE}" != "yes" ]; then
+  # If only Java 16 was available, then the first run would have used Java 16, so there's no reason to run again.
+  if [ "${has_java11}" = "yes" ] || [ "${has_java8}" = "yes" ]; then
```

Comment:
This block uses two `if` statements, but the logic for Java 11 uses one.  I think the code will be easier to read and understand if both blocks use logic of the same shape.